client/src/components/SearchBox/SearchBox.component.jsx

```diff
@@ -109,32 +121,38 @@ const SearchBox = ({ placeholder, value, handleSubmit }) => {
           }) => (
             <div
               className="search-autocomplete"
-              onBlur={() => sendAbandonedSearchEventToAnalytics(inputValue)}
+              onBlur={() => onAbandon(inputValue)}
             >
               <InputGroup className="search-box">
-                <InputLeftElement
-                  children={<BiSearch size="24" color="secondary.500" />}
-                  h="100%"
-                  w="24px"
-                  ml="19px"
-                  mr="20px"
-                />
+                {showSearchIcon ? (
+                  <InputLeftElement
+                    children={<BiSearch size="24" color="secondary.500" />}
+                    h="100%"
+                    w="24px"
+                    ml="19px"
+                    mr="20px"
+                  />
+                ) : null}
                 <Input
-                  variant="unstyled"
+                  variant={showSearchIcon ? 'unstyled' : undefined}
                   className="search-input"
-                  sx={{ paddingInlineStart: '63px' }}
+                  sx={{ paddingInlineStart: showSearchIcon ? '63px' : '16px' }}
                   {...getInputProps({
                     name,
                     placeholder,
-                    ...register(name),
+                    ref: inputRef,
+                    ...inputProps,
                     onKeyDown: (event) => {
-                      const selectingOptionUsingKeyboard =
-                        event.key === 'Enter' && highlightedIndex !== null
-                      if (selectingOptionUsingKeyboard) {
+                      if (event.key === 'Enter') {
                         // when selecting option using keyboard
                         // downshift prevents form submission which is used to submit analytics event
                         // detect such event and separately send analytics event
-                        sendSearchEventToAnalytics(inputValue)
+                        if (highlightedIndex === null && searchOnEnter) {
```

Comment:
nitpick: add comments explaining what the two cases cover. Its also a bit hard to understand because of the `===` and `!==`

Reply:
```suggestion
                        // hi
                        if (highlightedIndex === null && searchOnEnter) {
```

Reply:
```suggestion
                        // If user types enter without selecting anything
                        if (highlightedIndex === null && searchOnEnter) {
```